File: 1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        b_count = 0
        min_del = 0
        
        for i in s:
            if i == "a":
                min_del = min(min_del + 1, b_count)
            else:
                b_count += 1
                
        return min_del

File: 1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.java
class Solution {
    public int minimumDeletions(String s) {
        char[] arr = s.toCharArray();
        int count_b = 0;
        int min_del = 0;
        
        for (int i=0; i<arr.length; i++){
            if (arr[i] == 'a'){
                min_del = Math.min(min_del + 1, count_b);
            }
            else{
                count_b++;
            }
        }
        
        return min_del;
    }
}